### PR TITLE
Move pkg/api/ref.go and pkg/api/resource_helpers.go to subpackages

### PR DESCRIPTION
**Pull request by caesarxuchao**
First two commits are #44296 #44299. (unfortunately, removing these commits results in conflicts)

This PR moves resource_helpers.go to a subpackage. It's mostly a mechanic move, except that:
* i kept the methods of ResourceName and ResourceList in pkg/api/method.go

I'll send one more PR to separate api.Scheme etc. to their own package.

### CODE REVIEW

File: pkg/api/methods.go [hunk 1]
```diff
@@ -44,3 +49,47 @@ func (t *Taint) ToString() string {
...
+	obj.APIVersion, obj.Kind = gvk.ToAPIVersionAndKind()
+}
+func (obj *ObjectReference) GroupVersionKind() schema.GroupVersionKind {
+	return schema.FromAPIVersionAndKind(obj.APIVersion, obj.Kind)
+}
+
+func (obj *ObjectReference) GetObjectKind() schema.ObjectKind { return obj }
+
+// Returns string version of ResourceName.
+func (self ResourceName) String() string {
```
Comment (lavalamp) [inline pkg/api/methods.go]: same as comment on the other PR, but I'd rather these go in a resource-specific file?

File: pkg/api/resource/doc.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Package resource only exists until heapster rebases
+// Package resource has to exist until heapster rebases
```
Comment (lavalamp) [inline pkg/api/resource/doc.go]: no longer true, since you're moving things here, right?

Comment (caesarxuchao) [inline pkg/api/resource/doc.go]: Removed.

File: pkg/api/methods.go (hunk 1 above)
Comment (caesarxuchao) [inline pkg/api/methods.go]: Not sure what you meant by resource-specific file. I moved them to resource.go.

**Comment by caesarxuchao** [comment]
@lavalamp PTAL. Thanks.

File: pkg/api/resource/resource_helpers.go [hunk 3]
```diff
@@ -143,10 +113,10 @@ func GetNodeCondition(status *NodeStatus, conditionType NodeConditionType) (int,
 	return -1, nil
 }
 
-// Updates existing pod condition or creates a new one. Sets LastTransitionTime to now if the
+// UpdatePodCondition updates existing pod condition or creates a new one. Sets LastTransitionTime to now if the
```
Comment (lavalamp) [inline pkg/api/resource/resource_helpers.go]: I think these condition functions don't have anything to do with resources?

Comment (lavalamp) [inline pkg/api/resource/resource_helpers.go]: (sorry didn't notice this earlier)

Comment (caesarxuchao) [inline pkg/api/resource/resource_helpers.go]: Hmm, i'm just blindly moving the entire file, these functions are in the resource_helpers.go already: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/resource_helpers.go#L149.

Comment (caesarxuchao) [inline pkg/api/resource/resource_helpers.go]: I moved the pod utilities to pkg/api/pod, and removed the unused node utilities. There are two remaining functions that are related to "resource" in this file. PTAL. Thanks, @lavalamp :)

**Comment by lavalamp** [comment]
Great, thanks :)

/lgtm
/approve

**Pull request merged**